src/server/system_services/schemas/nsfs_bucket_schema.js

```diff
@@ -26,6 +25,7 @@ module.exports = {
         name: {
             type: 'string',
         },
+        // temp, keep the system_owner where it was already added (remove it in the future)
```

Comment:
@shirady @guymguym 
Are we all ok with not removing system_owner instead of upgrading the existing bucket config files?
Should we upgrade the files?

Reply:
Regarding removal for existing buckets - INHO we should have a single function in our codebase (in config_dir module or a bucketspace method...) that reads a bucket.json and it will immediately delete any old properties right there during load, so that then any read-modify-write flow that uses it, will always write the updated json without it.

Also we can use jsdoc @deprecated -
```suggestion
        /** 
         * @deprecated system_owner is kept for backward compatibility, 
         * but will no longer be included in new / updated bucket json.
         */
```

Reply:
@guymguym Regarding the deprecated tag - in the JSDOC example I can see that they mention since which version it was deprecated, don't we want to add it as well? ("deprecated since version 5.18").

A couple of questions regarding the suggestion to add the read-modify-write:
1. What about the entries in the `bucket_namespace_cache` that already have this property? Can we assume that the cache is empty because each node was upgraded?
2. The function `read_bucket_sdk_info` is used when listing the buckets - it would mean that we will read and write all the buckets on the first list-buckets, and I'm not sure we would like that.
3. In general - why not leave it read-modify and the write will happen only when updating a bucket?
cc: @romayalon

Reply:
sure, adding the version doesn't hurt.
1 - the cache is in memory, it will be cleared on upgrade.
2,3 - that's what I meant, we shouldn't write unless there is a need to update the bucket. in list/read flows we should only modify the object in memory so in case we need to write that it will be updated correctly, and also that any code that looks at this object in memory will see the correct strcture.

Reply:
I assumed the answers to this comment yesterday the code is updated - you can review it.
regarding the version - I added it in internal function  `adjust_bucket_with_schema_updates` and I don't want to write it twice, so it will appear there.